File: Data Structures/Array/LeetCode/217/217ContainsDuplicate.go
package main

func containsDuplicate(nums []int) bool {
	seen := make(map[int]bool)
	for _, num := range nums {
		if seen[num] {
			return true // Duplicate found
		}
		seen[num] = true // Mark number as seen
	}
	return false // No duplicates
}

// Example usage:
// func main() {
//     nums := []int{1, 2, 3, 1}
//     result := containsDuplicate(nums)
//     fmt.Println(result) // Output: true
// }

File: Data Structures/Array/LeetCode/217/217ContainsDuplicate.cpp
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (int num : nums) {
            if (seen.count(num)) {
                return true;  // Duplicate found
            }
            seen.insert(num);  // Add to the set
        }
        return false;  // No duplicates
    }
};

// Example usage:
// #include <iostream>
// int main() {
//     Solution solution;
//     vector<int> nums = {1, 2, 3, 1};
//     cout << solution.containsDuplicate(nums);  // Output: 1 (true)
// }